pinot-spi/src/main/java/org/apache/pinot/spi/stream/PartitionGroupMetadata.java

```diff
@@ -0,0 +1,76 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.spi.stream;
+
+/**
+ * A wrapper around all the information of a current Partition Group 
```

Comment:
Not really. This seems to be an extension of segment metadata. AFAICS there is a 1-1 correspondence between a segment and partition group metadata. What is not clear is whether it also exists for CONSUMING segments

Reply:
Added some clarifying comments. This is derived from the latest segment zk metadata of each partition group as found in the ideal state.